pkg/kudoctl/packages/verifier/template/verify_extended_parameters.go

```diff
@@ -0,0 +1,138 @@
+package template
+
+import (
+	"strings"
+
+	"github.com/kudobuilder/kudo/pkg/kudoctl/packages"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/verifier"
+)
+
+var _ packages.Verifier = &ExtendedParametersVerifier{}
+
+type ExtendedParametersVerifier struct {
+	VerifyParamDescription bool
+	VerifyParamHint        bool
+	VerifyParamDisplayName bool
+	VerifyParamType        bool
+	VerifyParamGroup       bool
+	VerifyGroups           bool
+}
+
+// Verify implements packages.Verifier for parameter verification
+func (v *ExtendedParametersVerifier) Verify(pf *packages.Files) verifier.Result {
+
+	res := verifier.NewResult()
+	if v.VerifyParamDescription {
+		res.Merge(v.verifyDescription(pf))
+	}
+	if v.VerifyParamHint {
+		res.Merge(v.verifyHint(pf))
+	}
+	if v.VerifyParamDisplayName {
+		res.Merge(v.verifyDisplayName(pf))
+	}
+	if v.VerifyParamGroup {
+		res.Merge(v.verifyParamGroup(pf))
+	}
+	if v.VerifyParamType {
+		res.Merge(v.verifyType(pf))
+	}
+	if v.VerifyGroups {
+		res.Merge(v.verifyGroups(pf))
+	}
+	return res
+}
+
+func (ExtendedParametersVerifier) verifyDescription(pf *packages.Files) verifier.Result {
+	res := verifier.NewResult()
+	for _, p := range pf.Params.Parameters {
+		if p.Description == "" {
+			res.AddParamWarning(p.Name, "has no description")
+		} else {
+			if !strings.HasSuffix(p.Description, ".") {
+				res.AddParamWarning(p.Name, "has a description not ending with a '.'")
+			}
+		}
+	}
+	return res
+}
+
+func (ExtendedParametersVerifier) verifyDisplayName(pf *packages.Files) verifier.Result {
+	res := verifier.NewResult()
+	for _, p := range pf.Params.Parameters {
+		if p.DisplayName == "" {
+			res.AddParamWarning(p.Name, "has no displayName")
```

Comment:
Do we have a use for the `displayName` now? I always thought that it is some poc that went nowhere and that it can be part of hint/description.

Reply:
It's part of the UI definition: 
- displayName -> Label
- hint -> Shows on-screen format hints, etc. (short one-liner)
- description -> Extended description, normally a (?) sign with mouse-over.